moped-editor/src/views/projects/projectView/ProjectNameEditable.js

```diff
@@ -1,221 +0,0 @@
-    cursor: "pointer",
-    marginTop: "16px",
-    fontSize: "2rem",
-  },
-  titleDisplayField: {
-    display: "flex",
-    cursor: "pointer",
-  },
-  titleEditField: {
-    "font-size": "1.5rem",
-    "font-weight": "bold",
-  },
-  boxField: {
-    width: "100%",
-  },
-  titleId: {
-    alignSelf: "center",
-  },
-}));
-
-const ProjectNameEditable = (props) => {
-  const classes = useStyles();
-
-  const DEFAULT_SNACKBAR_STATE = {
-    open: false,
-    message: null,
-    severity: "success",
-  };
-
-  const initialProjectName = props?.projectName;
-  const [projectName, setProjectName] = useState(initialProjectName);
-  const [projectNameBeforeEdit, setProjectNameBeforeEdit] =
-    useState(projectName);
-  const [isEditing, setIsEditing] = useState(props?.isEditing ?? false);
-  const [snackbarState, setSnackbarState] = useState(DEFAULT_SNACKBAR_STATE);
-  const [titleError, setTitleError] = useState(false);
-
-  if (props?.isEditing && !isEditing) {
-    setIsEditing(true);
-  }
-
-  const [updateProjectName] = useMutation(UPDATE_PROJECT_NAME_QUERY);
-
-  const handleProjectNameChange = (e) => {
-    if (titleError) {
-      setTitleError(false);
-    }
-    setProjectName(e.target.value);
-  };
-
-  /**
-   * Makes the update via GraphQL
-   */
-  const handleAcceptClick = (e) => {
-    e.preventDefault();
-    if (!(projectName.trim() === "")) {
-      updateProjectName({
-        variables: {
-          projectId: props?.projectId,
-          projectName: projectName,
-        },
-      })
-        .then((res) => {
-          setProjectNameBeforeEdit(projectName);
-          setSnackbarState({
-            open: true,
-            message: <span>Success! the project name has been updated!</span>,
-            severity: "success",
-          });
-        })
-        .catch((err) => {
-          setProjectName(projectNameBeforeEdit);
-          setSnackbarState({
-            open: true,
-            message: (
-              <span>There was a problem updating the project name.</span>
-            ),
-            severity: "error",
-          });
-          setProjectName(initialProjectName);
-        })
-        .finally(() => {
-          if (props?.setIsEditing) props.setIsEditing(false);
-          setIsEditing(false);
-          setTimeout(() => setSnackbarState(DEFAULT_SNACKBAR_STATE), 3000);
-          props.updatedCallback();
-        });
-    } else {
-      setTitleError(true);
-    }
-  };
-
-  /**
-   * Handles closing the edit mode
-   * @param {ChangeEvent} e - HTML DOM Event
-   */
-  const handleCancelClick = (e) => {
-    e.preventDefault();
-    if (props?.setIsEditing) props.setIsEditing(false);
-    setIsEditing(false);
-    setProjectName(projectNameBeforeEdit);
-  };
-
-  const handleSnackbarClose = () => {
-    setSnackbarState(DEFAULT_SNACKBAR_STATE);
-  };
-
-  return (
-    <Box className={isEditing ? classes.boxField : null}>
-      {isEditing && (
-        <form onSubmit={(e) => handleAcceptClick(e)}>
```

Comment:
we can sync about this tomorrow, I am going to try to compare this component to the way you have it set up now. Because as im looking at them now, they are pretty similar except for the extra text field, and removing the form tag

Reply:
ah, one thing the form tag offered, was being able to submit by hitting enter. I wonder if that functionality should stick around with the secondary name too.

Reply:
Thanks Chia! I definitely got to a point in this work where I was trying to reuse as much code as I could and started running into complications. At that point, I did change my approach, and began again, this time building from the ground up while referencing the old code. It makes sense to me that I may have introduced subtle changes, and I'll be happy to eliminate any regressions that I've introduced. Thank you - anytime you want to sync is great. :pray:

Reply:
I got that 'submit-on-enter' functionality restored. It's very strange - I had to wrap each input field's component in a `<form>` field individually, instead of wrapping the set of fields in a single enclosing form. Maybe this is a react thing?

At least it's working!